test/test_bible_routes/test_version_routes.py

```diff
@@ -98,38 +98,119 @@ def test_regular_user_create_list_and_delete_version(
 
 class TestAdminFlow:
     def test_admin_create_list_and_delete_version(
-        self, client, admin_token, regular_token1, db_session
+        self, client, admin_token, regular_token1, regular_token2, db_session
     ):
-        # Step 1: Create a version as an regular user
+        group_1 = db_session.query(Group).first()
+        assert group_1 is not None
+
+        # Create a second group for the first regular user
+        group_3 = Group(name="Group3", description="Test Group 3")
+        db_session.add(group_3)
+        db_session.commit()
+
+        user_1 = db_session.query(UserModel).filter_by(username="testuser1").first()
+        user_2 = db_session.query(UserModel).filter_by(username="testuser2").first()
+
+        # Associate both regular users with the new group
+        user_group3 = UserGroup(user_id=user_1.id, group_id=group_3.id)
+        user_group4 = UserGroup(user_id=user_2.id, group_id=group_3.id)
+        db_session.add_all([user_group3, user_group4])
+        db_session.commit()
+
+        # Assert user 1 has access to group 1
+        user_group = (
+            db_session.query(UserGroup)
+            .filter(UserGroup.user_id == user_1.id, UserGroup.group_id == group_1.id)
+            .first()
+        )
+        assert user_group is not None
+
+        # Assert user_2 does not have access to group 1
+        user_group = (
+            db_session.query(UserGroup)
+            .filter(UserGroup.user_id == user_2.id, UserGroup.group_id == group_1.id)
+            .first()
+        )
+        assert user_group is None
+
+        # Assert both users have access to group 3
+        user_group = (
+            db_session.query(UserGroup)
+            .filter(UserGroup.user_id == user_1.id, UserGroup.group_id == group_3.id)
+            .first()
+        )
+        assert user_group is not None
+        user_group = (
+            db_session.query(UserGroup)
+            .filter(UserGroup.user_id == user_2.id, UserGroup.group_id == group_3.id)
+            .first()
+        )
+        assert user_group is not None
+
```

Comment:
i like the db setup here, i may copy for future reference, with this flow and the assertions the developer never has to look back at the conftest file